pkg/inject/server.go

```diff
@@ -54,43 +55,55 @@ type WebhookServer struct {
 
 // Webhook Server parameters
 type WebhookServerParameters struct {
+	NoTLS    bool   // disables TLS
 	Port     int    // webhook Server port
 	CertFile string // path to the x509 certificate for https
 	KeyFile  string // path to the x509 private key matching `CertFile`
 }
 
-func failWithResponse(errMsg string) *v1beta1.AdmissionResponse {
-	glog.Infof(errMsg)
-	return &v1beta1.AdmissionResponse{
+func failWithResponse(errMsg string) v1beta1.AdmissionResponse {
+	log.Printf(errMsg)
+	return v1beta1.AdmissionResponse{
 		Result: &metav1.Status{
 			Message: errMsg,
 		},
 	}
 }
 
-// main mutation process
-func (whsvr *WebhookServer) mutate(ar *v1beta1.AdmissionReview) *v1beta1.AdmissionResponse {
-	req := ar.Request
+// HandleAdmissionRequest applies the sidecar-injector logic to the AdmissionRequest
+// and returns the results as an AdmissionResponse.
+func HandleAdmissionRequest(req *v1beta1.AdmissionRequest) v1beta1.AdmissionResponse {
+	if req == nil {
+		return failWithResponse("Received empty request")
+	}
+
 	var pod corev1.Pod
 	if err := json.Unmarshal(req.Object.Raw, &pod); err != nil {
 		return failWithResponse(fmt.Sprintf("Could not unmarshal raw object: %v", err))
 	}
 
-	glog.Infof("AdmissionReview for Kind=%v, Namespace=%v Name=%v (%v) UID=%v rfc6902PatchOperation=%v UserInfo=%v",
-		req.Kind, req.Namespace, req.Name, pod.Name, req.UID, req.Operation, req.UserInfo)
+	log.Printf("AdmissionReview for Kind=%v, Namespace=%v PodName=%v UID=%v rfc6902PatchOperation=%v UserInfo=%v",
+		req.Kind, req.Namespace, metaName(&pod.ObjectMeta), req.UID, req.Operation, req.UserInfo)
 
 	// determine whether to perform mutation
 	if !mutationRequired(ignoredNamespaces, &pod.ObjectMeta) {
-		glog.Infof("Skipping mutation for %s/%s due to policy check", req.Namespace, pod.Name)
-		return &v1beta1.AdmissionResponse{
+		log.Printf("Skipping mutation for %s/%s due to policy check", req.Namespace, pod.Name)
+		return v1beta1.AdmissionResponse{
 			Allowed: true,
 		}
 	}
 
 	injectType, err := getAnnotation(&pod.ObjectMeta, annotationInjectTypeKey)
 	containerMode, err := getAnnotation(&pod.ObjectMeta, annotationContainerModeKey)
 	containerName, err := getAnnotation(&pod.ObjectMeta, annotationContainerNameKey)
-
+	conjurTokenReceiversStr, err := getAnnotation(
+		&pod.ObjectMeta,
+		annotationConjurTokenReceiversKey,
+	)
+	conjurTokenReceivers := strings.Split(conjurTokenReceiversStr, ",")
```

Comment:
Allow a space after the comma
```suggestion
	conjurTokenReceivers := regexp.MustCompile(", ?").Split(conjurTokenReceiversStr, -1)
```

Reply:
Hah, just realised this has no effect, because you're trimming off spaces anyway.